controllers/storagecluster/odfinfoconfig.go

```diff
@@ -0,0 +1,231 @@
+	return sc.Namespace + "/" + sc.Name + "." + OdfInfoKeyName
+}
+
+// ensureDeleted is dummy func for the odfInfoConfig
+func (obj *odfInfoConfig) ensureDeleted(r *StorageClusterReconciler, sc *ocsv1.StorageCluster) (reconcile.Result, error) {
+	operatorNamespace, err := util.GetOperatorNamespace()
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+	var odfInfoConfigMap corev1.ConfigMap
+	if err = r.Client.Get(r.ctx, types.NamespacedName{Namespace: operatorNamespace}, &odfInfoConfigMap); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	mutex.Lock()
+	defer mutex.Unlock()
+	if len(odfInfoConfigMap.Data) > 1 {
+		delete(odfInfoConfigMap.Data, obj.getOdfInfoKeyName(sc))
+	} else {
+		if err = r.Client.Delete(r.ctx, &odfInfoConfigMap); err != nil {
+			r.Log.Error(err, "Failed to delete odf-info.", "ConfigMap", klog.KRef(odfInfoConfigMap.Namespace, odfInfoConfigMap.Name))
+			return reconcile.Result{}, fmt.Errorf("failed to delete odf-info %v: %v", odfInfoConfigMap.Name, err)
+		}
+	}
+	return reconcile.Result{}, nil
+}
+
+func getOdfInfoData(r *StorageClusterReconciler, sc *ocsv1.StorageCluster) (string, error) {
+	var odfVersion, cephFSId string
+	var err error
+	if odfVersion, err = getOdfVersion(r, sc); err != nil {
+		return "", err
+	}
+	if cephFSId, err = getCephFsid(r, sc); err != nil {
+		return "", err
+	}
+	var odfDeploymentType string
+	if sc.Spec.ExternalStorage.Enable {
+		odfDeploymentType = OdfDeploymentTypeExternal
+	} else {
+		odfDeploymentType = OdfDeploymentTypeInternal
+	}
+	var isDROptimized = false
+	// Set isDROptmized to "false" in case of external clusters as we currently don't have to way to determine
+	// if external cluster OSDs are using bluestore-rdr
+	if !sc.Spec.ExternalStorage.Enable {
+		if isDROptimized, err = getIsDROptimized(r, sc); err != nil {
+			return "", err
+
+		}
+	}
+	var storageSystemName string
+	if storageSystemName, err = getStorageSystemName(sc); err != nil {
+		return "", err
+	}
+	storageClusterCount := len(r.clusters.GetStorageClusters())
+
+	var data = OdfInfoData{
+		OdfVersion:          odfVersion,
+		OdfDeploymentType:   odfDeploymentType,
+		StorageClusterCount: storageClusterCount,
+		IsDROptimized:       isDROptimized,
+		StorageSystemName:   storageSystemName,
+		// Clients array is populated with the onboarding request's fields via server.go
+		Clients: []ConnectedClient{},
+		StorageCluster: InfoStorageCluster{
+			NamespacedName:          types.NamespacedName{Name: sc.Name, Namespace: sc.Namespace},
+			StorageProviderEndpoint: sc.Status.StorageProviderEndpoint,
+			CephClusterFSID:         cephFSId,
+		},
+	}
+	yamlData, err := yaml.Marshal(data)
+	if err != nil {
+		return "", err
+	}
+	return string(yamlData), nil
+
+}
+
+func getStorageSystemName(storageCluster *ocsv1.StorageCluster) (string, error) {
+	for i := range storageCluster.OwnerReferences {
+		ref := &storageCluster.OwnerReferences[i]
+		if ref.Kind == "StorageSystem" {
+			return ref.Name, nil
+		}
+	}
```

Comment:
Using the recently added `Find` util, this can be simplified to:

```suggestion
	storageSystemRef := utils.Find(storageCluster.OwnerReferences, func (ref *metav1.OwnerReference) bool {
		return ref.Kind == "StorageSystem"
	})
	if storageSystemRef!= nil {
		return storageSystemRef.Name, nil
	}
```